src/blob/file.rs

```diff
@@ -93,35 +99,42 @@ impl File {
     async fn write_at_sync(&self, offset: u64, buf: &[u8]) -> IOResult<()> {
         let buf = buf.to_vec();
         let file = self.no_lock_fd.clone();
-        Self::blocking_call(move || file.write_all_at(&buf, offset)).await
+        if buf.len() <= MAX_SYNC_OPERATION_SIZE {
+            Self::inplace_sync_call(move || file.write_all_at(&buf, offset))
+        } else {
+            Self::background_sync_call(move || file.write_all_at(&buf, offset)).await
+        }
     }
 
-    pub(crate) async fn read_all(&self) -> Result<Vec<u8>> {
-        let mut buf = vec![0; self.size().try_into()?];
-        self.read_at(&mut buf, 0).await?; // TODO: verify read size
-        Ok(buf)
+    pub(crate) async fn read_all(&self) -> Result<BytesMut> {
+        self.read_at_allocate(self.size().try_into()?, 0).await
     }
 
-    pub(crate) async fn read_at(&self, buf: &mut [u8], offset: u64) -> Result<()> {
+    pub(crate) async fn read_at_allocate(&self, size: usize, offset: u64) -> Result<BytesMut> {
```

Comment:
I think it would be good to add `exact` word to the name: `read_exact_at_allocate`